src/Console/Command/DumpMergesCommand.php

```diff
@@ -187,6 +194,31 @@ private function getSinceIdFromInputAndContent(InputInterface $input, string $co
             return (int) $sinceId;
         }
 
+        $baseBranch = $input->getOption(Option::BASE_BRANCH);
+
+        if ($baseBranch !== null) {
+            return $this->findHighestIdMergedInBranch($content, $baseBranch);
+        }
+
         return $this->idsAnalyzer->getHighestIdInChangelog($content);
     }
+
+    /**
+     * @param string $content
+     * @param string $branch
+     * @return int|null
+     */
+    private function findHighestIdMergedInBranch(string $content, string $branch): ?int
+    {
+        $allIdsInChangelog = $this->idsAnalyzer->getAllIdsInChangelog($content);
+        rsort($allIdsInChangelog);
+
+        foreach ($allIdsInChangelog as $id) {
+            if ($this->githubApi->isPullRequestMergedToBaseBranch((int) $id, $branch)) {
+                return (int) $id;
```

Comment:
You are using `$id` as int on both places so why don't you change the type of it before the `if`?

Reply:
yes, good point, I have added it in a new commit, I will need to release new patch version after you approve it :wink: